### ISSUE TITLE
FastaiLRFinder does not run more than 1 epoch. Why?

### ISSUE THREAD

**Issue by rafaelpadilla** [pytorch/ignite#1357]
I have been trying to use the `FastaiLRFinder` to find the best learning rate for my module.

If we create the trainer with the function `create_supervised_trainer` as below:

```
trainer = create_supervised_trainer(
    model, optimizer, criterion, device, output_transform=custom_output_transform
)
```

and run it:

```
with lr_finder.attach(
            trainer,
            to_save=to_save,
            num_iter=50,  
            end_lr=1.,
            step_mode='exp') as lr_finder_training:
        lr_finder_training.run(train_loader)
```

A warning will come up say: "UserWarning: Desired num_iter 50 is unreachable with the current run setup of 15 iteration (1 epochs) from ignite.contrib.handlers.param_scheduler import (LRScheduler, PiecewiseLinear)"

My dataloader has 15 batches to iterate, which means FastaiLRFinder does not allow you to run more than 1 epoch. Why?

According to their source code, we can see [here](https://github.com/pytorch/ignite/blob/14cdc046c7cac2db8a1bf838dcd09cbbe0e9c33b/ignite/contrib/handlers/lr_finder.py#L87) that this verification limits the user to run more iterations than the maximum of epochs in their dataloader.

But why? Am I missing something important here?

**Comment by sdesrozis** [comment]
@rafaelpadilla Thank you for this question.

I agree with you, I don't see any reason to limit the run to maximum epoch. Maybe @vfdev-5 could have more inputs about it.

**Comment by vfdev-5** [comment]
@rafaelpadilla you can either set epoch length to more than 50 iterations or add more epochs to run:
```python
lr_finder_training.run(train_loader, epoch_length=100)
# or 
# lr_finder_training.run(train_loader, max_epochs=10)
```

**Comment by rafaelpadilla** [comment]
> @rafaelpadilla you can either set epoch length to more than 50 iterations or add more epochs to run:
> 
> ```python
> lr_finder_training.run(train_loader, epoch_length=100)
> # or 
> # lr_finder_training.run(train_loader, max_epochs=10)
> ```
> 
> **EDIT:** I agree that the API should encounter this situation and either warn the user about that or set automatically `epoch_length` to `num_iter`...

@vfdev-5 
It can be solved by setting `epoch_length`. 

But I guess it makes the `num_iter` parameter passed by the `attach` method meaningless.

The API is confusing.

**Comment by vfdev-5** [comment]
@KickItLikeShika you have worked on LR finder recently. Can I assign this issue to you to improve it ?

**Comment by KickItLikeShika** [comment]
@vfdev-5 Yes that's a good one, I can work on it, thanks!
